server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -2,37 +2,108 @@ package queue
 
 import (
 	"container/list"
+	"fmt"
 
+	activity "github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
 )
 
-// Queue is a standard queue implementation
-type Queue struct {
-	queue *list.List
+type LockStatus int
+
+const (
+	UnlockedStatus LockStatus = iota
+	LockedStatus
+)
+
+type Deploy struct {
+	queue *priority
+
+	// mutable: default is unlocked
+	lock LockStatus
 }
 
-func NewQueue() *Queue {
-	return &Queue{
-		queue: list.New(),
+func NewQueue() *Deploy {
+	return &Deploy{
+		queue: newPriorityQueue(),
 	}
+
+}
+
+func (q *Deploy) SetLockStatusForMergedTrigger(status LockStatus) {
+	q.lock = status
 }
 
-func (q *Queue) IsEmpty() bool {
-	return q.queue.Len() == 0
+func (q *Deploy) CanPop() bool {
+	return q.lock == LockedStatus && q.queue.HasItemsOfPriority(High) || (q.lock == UnlockedStatus && !q.queue.IsEmpty())
```

Comment:
Can we simplify this conditional to: 
```
q.queue.HasItemsOfPriority(High) || (q.lock == UnlockedStatus && !q.queue.IsEmpty())
```
since we don't have to worry about the lock status if we have HP items in the queue right?

Reply:
yeah that's true, I'll simplify